gorm.go

```diff
@@ -347,6 +347,10 @@ func (db *DB) Callback() *callbacks {
 
 // AddError add error to db
 func (db *DB) AddError(err error) error {
+	if errTranslator, ok := db.Dialector.(ErrorTranslator); ok {
```

Comment:
We don't need to convert when `err == nil`, are you interested in creating an other PR for it?

Reply:
Yes sure